taichi/codegen/cuda/codegen_cuda.cpp

```diff
@@ -161,6 +162,39 @@ class TaskCodeGenCUDA : public TaskCodeGenLLVM {
     llvm_val[stmt] = create_print(formats, types, values);
   }
 
+  void visit(AllocaStmt *stmt) override {
+    // Override shared memory codegen logic for large shared memory
+    if (stmt->ret_type->is<TensorType>() && stmt->is_shared) {
+      auto tensor_type = stmt->ret_type->cast<TensorType>();
+      size_t shared_array_threshold = 49152;
```

Comment:
shall we move this to taichi/inc/constants.h and make it a constexpr?

Reply:
or make it a compile config?